Branch-SDK/src/io/branch/referral/RemoteInterface.java

```diff
@@ -26,128 +27,183 @@
+        try {
+            if (url.indexOf('?') == -1) {
+                url += "?sdk=android" + SDK_VERSION;
+            } else {
+                url += "&sdk=android" + SDK_VERSION;
+            }
+            if (log) {
+                PrefHelper.Debug(BranchConfig.TAG, "getting " + url);
+            }
+            HttpGet request = new HttpGet(url);
+            HttpResponse response = getGenericHttpClient(timeout).execute(request);
+
+            int statusCode = response.getStatusLine().getStatusCode();
+            boolean isRetryableStatusCode = statusCode >= 500;
+
+            // Retry request before processing
+            if ( (retryNumber < prefHelper_.getRetryCount()) && isRetryableStatusCode) {
+                retryNumber +=1;
+                make_restful_get(url, tag, timeout, retryNumber, log);
+            } else {
+                return processEntityForJSON(response.getEntity(),
+                        response.getStatusLine().getStatusCode(), tag, log, null);
+            }
+
+        } catch (ClientProtocolException ex) {
+            if (log) {
+                PrefHelper.Debug(getClass().getSimpleName(), "Client protocol exception: " + ex.getMessage());
+            }
+        } catch (SocketException ex) {
+            if (log) {
+                PrefHelper.Debug(getClass().getSimpleName(), "Http connect exception: " + ex.getMessage());
+            }
+            return new ServerResponse(tag, NO_CONNECTIVITY_STATUS);
+        } catch (UnknownHostException ex) {
+            if (log) {
+                PrefHelper.Debug(getClass().getSimpleName(), "Http connect exception: " + ex.getMessage());
+            }
+            return new ServerResponse(tag, NO_CONNECTIVITY_STATUS);
+        } catch (IOException ex) {
+            if (log) {
+                PrefHelper.Debug(getClass().getSimpleName(), "IO exception: " + ex.getMessage());
+            }
+            return new ServerResponse(tag, 500);
+        }
+        return null;
+    }
+    //endregion
+
+    //region POST requests
+    public ServerResponse make_restful_post(JSONObject body, String url, String tag, int timeout) {
+        return make_restful_post(body, url, tag, timeout, 0, true, null);
+    }
+
+    public ServerResponse make_restful_post(JSONObject body, String url, String tag, int timeout, BranchLinkData linkData) {
+        return make_restful_post(body, url, tag, timeout, 0, true, linkData);
+    }
+
+    public ServerResponse make_restful_post(JSONObject body, String url, String tag, int timeout, boolean log) {
+        return make_restful_post(body, url, tag, timeout, 0, log, null);
+    }
+
+    public ServerResponse make_restful_post(JSONObject body, String url, String tag, int timeout,
+                                            int retryNumber, boolean log, BranchLinkData linkData) {
+        try {
+            if (body.has("app_id") && body.getString("app_id").equals(PrefHelper.NO_STRING_VALUE)) {
```

Comment:
This is all super valuable feedback but it actually doesn't pertain to what
I changed. Let's sync up about this tomorrow? So I can go over with you two
what I changed then we finalize?

On Thursday, April 9, 2015, qinweigong <notifications@github.com> wrote:

> In Branch-SDK/src/io/branch/referral/RemoteInterface.java
> <https://github.com/BranchMetrics/Branch-Android-SDK/pull/58#discussion_r28121046>
> :
>
> > +    public ServerResponse make_restful_post(JSONObject body, String url, String tag, int timeout) {
> > +        return make_restful_post(body, url, tag, timeout, 0, true, null);
> > +    }
> > +
> > +    public ServerResponse make_restful_post(JSONObject body, String url, String tag, int timeout, BranchLinkData linkData) {
> > +        return make_restful_post(body, url, tag, timeout, 0, true, linkData);
> > +    }
> > +
> > +    public ServerResponse make_restful_post(JSONObject body, String url, String tag, int timeout, boolean log) {
> > +        return make_restful_post(body, url, tag, timeout, 0, log, null);
> > +    }
> > +
> > +    public ServerResponse make_restful_post(JSONObject body, String url, String tag, int timeout,
> > +                                            int retryNumber, boolean log, BranchLinkData linkData) {
> > +        try {
> > +            if (body.has("app_id") && body.getString("app_id").equals(PrefHelper.NO_STRING_VALUE)) {
>
> @shortstuffsushi <https://github.com/shortstuffsushi> I am make that
> change in my PR, FYI.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BranchMetrics/Branch-Android-SDK/pull/58/files#r28121046>
> .
>


-- 

*Sahil Verma*

*Branch Metrics *

w branch.io | c 510-386-1750 | s sahilroyverma